### PR TITLE
deploy api image via github actions

### PR DESCRIPTION
**Pull request by dehli**
Let me know if you'd like any changes. Builds and deploys api on each push to `master` and `prod`. Successful run can be seen [here](https://github.com/daimo-eth/daimo/actions/runs/9225477659/job/25383053568).

### CODE REVIEW

File: .github/workflows/cd.yml [hunk 1]
```diff
@@ -2,10 +2,61 @@ name: CD
...
+        with:
+          platforms: arm64
+
+      - name: Build, tag, and push docker image to Amazon ECR
+        env:
+          GIT_SHA: ${{ github.sha }}
+          IMAGE_NAME: ${{ steps.login-ecr.outputs.registry }}/${{ vars.AWS_ECR_API_REPO }}
+        run: |
+          docker build --platform linux/arm64 -t $IMAGE_NAME:$GIT_SHA -t $IMAGE_NAME:$IMAGE_TAG -f packages/daimo-api/Dockerfile .
+          docker push $IMAGE_NAME --all-tags
```
Comment (dcposch) [inline .github/workflows/cd.yml:51]: do we want `--all-tags` or just stage/prod tag, depending on branch?

**Review by dcposch — APPROVED** [review]
otherwise LGTM

File: .github/workflows/cd.yml (hunk 1 above)
Comment (dehli) [reply .github/workflows/cd.yml:51]: With the current code we tag the SHA in addition to latest/prod (depending on the branch) and push both tags up.

image

Would you prefer to just have the single latest/prod tag @dcposch?

Comment (dcposch) [reply .github/workflows/cd.yml:51]: ah, i see.  that looks good

Comment (dcposch) [reply .github/workflows/cd.yml:51]: as long as we're sure that a `master` build will never overwrite the `prod` tag

Comment (dehli) [reply .github/workflows/cd.yml]: The action's container will always  run in a fresh state so there won't be more than those two tags however I went ahead and updated the command to be more explicit for clarity.

**Pull request merged**